### PR TITLE
[expo-updates] add TS implementation, jest tests, and web shim

### LINKED ISSUE #6517: Updates unimodule

### PR DESCRIPTION
**Pull request by esamelson**
# Why

partial step of #6517

# How

- Brought package.json & rest of module up-to-date with other unimodules, including bumping `expo-module-scripts` version
- copied the meat of the TS implementation from the current `Updates.ts` code in the `expo` package as it's very similar, and made changes where necessary

# Test Plan

- [x] Manual testing
- [x] Maybe add a few jest tests

### CODE REVIEW

File: packages/expo-updates/src/__tests__/Updates-test.native.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import ExpoUpdates from '../ExpoUpdates';
+
+const fakeManifest = {
+  id: '@jester/test-app',
+  sdkVersion: '36.0.0',
+};
+
+it('returns the proper object in checkForUpdateAsync if an update is available and manifest is returned', async () => {
+  const mock = jest.fn();
+  ExpoUpdates.checkForUpdateAsync = mock;
```
Comment (ide) [inline packages/expo-updates/src/__tests__/Updates-test.native.ts]: Shouldn't this already be set to a mock function by the Jest preset? If the intent here is to clear the mock's state, call `ExpoUpdates.checkForUpdateAsync.mockReset()` in afterEach.

**Review by ide — CHANGES_REQUESTED** [review]
Just some small changes.

File: packages/expo-updates/src/Updates.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  if (!_emitter) {
+    _emitter = new EventEmitter();
+    RCTDeviceEventEmitter.addListener('Expo.nativeUpdatesEvent', _emitEvent);
+  }
+  return _emitter;
+}
+
+function _emitEvent(params): void {
+  let newParams = params;
+  if (typeof params === 'string') {
```
[the remark this answers is not in the document]
Comment (esamelson) [reply packages/expo-updates/src/Updates.ts]: yes -- the manifest is a `JSONObject` in Java but afaict there is no natural way to convert `JSONObject` into an object readable by RN on the native side

[the remark this answers is not in the document]
Comment (ide) [reply packages/expo-updates/src/Updates.ts]: OK, will be nice to replace this with Turbomodules one day (no more emitters either)

**Review by ide — APPROVED** [review]

**Pull request merged**